backend/core/autoignore.py

```diff
@@ -0,0 +1,350 @@
+            except Exception:
+                log.exception("Exception")
+            finally:
+                signal_loading("autoignore", False)
+
+        def config_request_rpc(self):
+            self.correlation_id = uuid()
+            callback_queue = Queue(
+                uuid(),
+                durable=False,
+                auto_delete=True,
+                max_priority=4,
+                consumer_arguments={"x-priority": 4},
+            )
+
+            self.producer.publish(
+                "",
+                exchange="",
+                routing_key="config-request-queue",
+                reply_to=callback_queue.name,
+                correlation_id=self.correlation_id,
+                retry=True,
+                declare=[
+                    Queue(
+                        "config-request-queue",
+                        durable=False,
+                        max_priority=4,
+                        consumer_arguments={"x-priority": 4},
+                    ),
+                    callback_queue,
+                ],
+                priority=4,
+                serializer="ujson",
+            )
+            with Consumer(
+                self.connection,
+                on_message=self.handle_config_request_reply,
+                queues=[callback_queue],
+                accept=["ujson"],
+            ):
+                while self.autoignore_rules is None:
+                    self.connection.drain_events()
+
+        def handle_config_request_reply(self, message):
+            message.ack()
+            signal_loading("autoignore", True)
+            log.debug("message: {}\npayload: {}".format(message, message.payload))
+            if self.correlation_id == message.properties["correlation_id"]:
+                config = message.payload
+                if config["timestamp"] > self.timestamp:
+                    self.timestamp = config["timestamp"]
+                    self.autoignore_rules = config.get("autoignore", [])
+                    self.build_prefix_tree()
+                    self.set_rule_timers()
+            signal_loading("autoignore", False)
+
+        def build_prefix_tree(self):
+            log.info("Starting building autoignore prefix tree...")
+            self.prefix_tree = {
+                "v4": pytricia.PyTricia(32),
+                "v6": pytricia.PyTricia(128),
+            }
+            raw_prefix_count = 0
+            for key in self.autoignore_rules:
+                try:
+                    rule = self.autoignore_rules[key]
+                    for prefix in rule["prefixes"]:
+                        for translated_prefix in translate_rfc2622(prefix):
+                            ip_version = get_ip_version(translated_prefix)
+                            if self.prefix_tree[ip_version].has_key(translated_prefix):
+                                node = self.prefix_tree[ip_version][translated_prefix]
+                            else:
```

Comment:
@slowr: this is what happens when the same prefix is in multiple rules. Could you let me know what should be the correct behavior? (instead of the current overwriting that is effectively done)

Reply:
actually check also my comment below, this overwriting does not hurt (it is OK performance-wise, since we examine rules one-by-one when the timers reset)